plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -270,6 +282,72 @@ private void testAddNestedColumnWithColumnMappingMode(String columnMappingMode)
+    void testPartitionValuesParsedCheckpoint()
+            throws Exception
+    {
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.ID);
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.NAME);
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.NONE);
+    }
+
+    private void testPartitionValuesParsedCheckpoint(ColumnMappingMode columnMappingMode)
+            throws Exception
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_checkpoint",
+                "(x int, part int) WITH (checkpoint_interval = 3, column_mapping_mode = '" + columnMappingMode + "', partitioned_by = ARRAY['part'])")) {
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (1, 10)", 1);
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (2, 20)", 1);
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (3, 30)", 1);
+
+            Path tableLocation = Path.of(getTableLocation(table.getName()).replace("file://", ""));
+            Path checkpoint = tableLocation.resolve("_delta_log/00000000000000000003.checkpoint.parquet");
+
+            MetadataEntry metadataEntry = loadMetadataEntry(0, tableLocation);
+            ProtocolEntry protocolEntry = loadProtocolEntry(0, tableLocation);
+
+            DeltaLakeColumnHandle partitionColumn = extractPartitionColumns(metadataEntry, protocolEntry, TESTING_TYPE_MANAGER).stream().collect(onlyElement());
+            String physicalColumnName = partitionColumn.basePhysicalColumnName();
+            if (columnMappingMode == ColumnMappingMode.ID || columnMappingMode == ColumnMappingMode.NAME) {
+                assertThat(physicalColumnName).matches(PHYSICAL_COLUMN_NAME_PATTERN);
+            }
+            else {
+                assertThat(physicalColumnName).isEqualTo("part");
+            }
+
+            int partitionValuesParsedFieldPosition = 6;
+            RowType addEntryType = new CheckpointSchemaManager(TESTING_TYPE_MANAGER).getAddEntryType(metadataEntry, protocolEntry, _ -> true, true, true, true);
+
+            RowType.Field partitionValuesParsedField = addEntryType.getFields().get(partitionValuesParsedFieldPosition);
+            assertThat(partitionValuesParsedField.getName().orElseThrow()).matches("partitionValues_parsed");
+            RowType partitionValuesParsedType = (RowType) partitionValuesParsedField.getType();
+            assertThat(partitionValuesParsedType.getFields().stream().collect(onlyElement()).getName().orElseThrow()).isEqualTo(physicalColumnName);
+
+            ParquetMetadata parquetMetadata = MetadataReader.readFooter(
+                    new TrinoParquetDataSource(new LocalInputFile(checkpoint.toFile()), new ParquetReaderOptions(), new FileFormatDataSourceStats()),
+                    Optional.empty());
+            ParquetDataSource dataSource = new FileParquetDataSource(checkpoint.toFile(), new ParquetReaderOptions());
```

Comment:
I think you can use the same dataSource in MetadataReader.readFooter as well as createParquetReader